src/JSInterop/Microsoft.JSInterop/src/TaskGenericsUtil.cs

```diff
@@ -25,20 +24,24 @@ public static void SetTaskCompletionSourceException(object taskCompletionSource,
         public static Type GetTaskCompletionSourceResultType(object taskCompletionSource)
             => CreateResultSetter(taskCompletionSource).ResultType;
 
-        public static object GetTaskResult(Task task)
+        public static object? GetTaskResult(Task task)
         {
-            var getter = _cachedResultGetters.GetOrAdd(task.GetType(), taskInstanceType =>
+            var taskInstanceType = task.GetType();
+            if (!_cachedResultGetters.TryGetValue(taskInstanceType, out var getter))
             {
                 var resultType = GetTaskResultType(taskInstanceType);
-                return resultType == null
-                    ? new VoidTaskResultGetter()
-                    : (ITaskResultGetter)Activator.CreateInstance(
+                getter = resultType == null ?
```

Comment:
A bit unrelated to reference types, but this would avoids the delegate allocation